tests/wallet/nft_wallet/test_nft_wallet.py

```diff
@@ -673,21 +666,13 @@ async def test_nft_with_did_wallet_creation(two_wallet_nodes: Any, trusted: Any)
     assert compute_memos(sb)
     await time_out_assert_not_none(5, full_node_api.full_node.mempool_manager.get_spendbundle, sb.name())
 
-    for i in range(1, num_blocks):
-        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
-    await time_out_assert(10, wallet_0.get_unconfirmed_balance, 13999999999998 - 1)
-    await time_out_assert(10, wallet_0.get_confirmed_balance, 13999999999998 - 1)
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+    await time_out_assert(10, wallet_0.get_unconfirmed_balance, 7999999999998 - 1)
+    await time_out_assert(10, wallet_0.get_confirmed_balance, 7999999999998 - 1)
     # Check DID NFT
-    time_left = 5.0
-    coins_response = {}
-    while time_left > 0:
-        coins_response = await api_0.nft_get_nfts(dict(wallet_id=nft_wallet_0_id))
-        if coins_response.get("nft_list"):
-            break
-        await asyncio.sleep(0.5)
-        time_left -= 0.5
-    assert coins_response["nft_list"], isinstance(coins_response, dict)
-    assert coins_response.get("success")
+    coins_response = await wait_rpc_state_condition(
+        5, api_0.nft_get_nfts, [dict(wallet_id=nft_wallet_0_id)], lambda x: x["nft_list"]
```

Comment:
I'm guessing we mean something more like `len(x["nft_list"]) > 0`?  We probably don't want to succeed in cases where `x["nft_list"]` is `object()` or `True` or `7`.